### PR TITLE
fix: Use dao type script hash in dao transaction

### PR DESCRIPTION
**Pull request by TheWaWaR**
Dependent on this PR https://github.com/nervosnetwork/ckb/pull/1600

### CODE REVIEW

File: test/src/node.rs [hunk 1]
```diff
@@ -330,7 +330,7 @@ impl Node {
         since: u64,
         capacity: Capacity,
     ) -> TransactionView {
-        let always_success_out_point = OutPoint::new(self.genesis_cellbase_hash.clone(), 1);
+        let always_success_out_point = OutPoint::new(self.genesis_cellbase_hash.clone(), 5);
```
Comment (u2) [inline test/src/node.rs]: Use `SYSTEM_CELL_ALWAYS_SUCCESS_INDEX` instead of `5`

File: test/src/node.rs [hunk 2]
```diff
@@ -384,7 +388,7 @@ impl Node {
         self.always_success_code_hash = CellOutput::calc_data_hash(
             &consensus.genesis_block().transactions()[0]
                 .outputs_data()
-                .get(1)
+                .get(5)
```
Comment (u2) [inline test/src/node.rs]: SYSTEM_CELL_ALWAYS_SUCCESS_INDEX

**Comment by u2** [comment]
bors r=quake,u2

File: spec/src/consensus.rs [hunk 3]
```diff
@@ -227,8 +231,27 @@ impl ConsensusBuilder {
...
+
+        let genesis_cellbase = genesis_block
+            .transaction(0)
+            .expect("Genesis must have cellbase");
+        let get_type_hash = |output_index: u64| {
+            genesis_cellbase
+                .output(output_index as usize)
+                .and_then(|output| output.type_().to_opt())
+                .map(|type_script| type_script.calc_script_hash())
+                .unwrap_or_default()
```
Comment (keroro520) [inline spec/src/consensus.rs]: Better `expect` here but not return a wrong hash value.

**Comment by keroro520** [comment]
@TheWaWaR resolve conflict

**Pull request merged**